### ISSUE TITLE
Which feature is needed for Document Symbol Request

### ISSUE THREAD

**Issue by nilsmahlstaedt** [langserver/langserver.github.io#183]
Hi,
I have a short question regarding the capabilities of the language servers:

Which feature is needed to be able to use the [Document Symbol Request](https://microsoft.github.io/language-server-protocol/specifications/specification-3-14/#textDocument_documentSymbol) ?

Is it `Jump to def` or does the server actually have to support `Workspace Symbols`?

Thank you for keeping up this helpful list!
Nils

**Comment by keegancsmith** [comment]
This is for showing the symbols in a file. For example in vscode it is in the outline view https://code.visualstudio.com/docs/getstarted/userinterface#_outline-view

**Comment by nilsmahlstaedt** [comment]
Hi @keegancsmith thank you for your swift answer!
I probably did not phrase that question the right way, sorry for that.
Let me try again:

On https://langserver.org/#implementations-server you have a table that lists language server implementations and the functionalities these implementations support.
Can I infer from that table which servers are likely to support the  
[Document Symbol Request](https://microsoft.github.io/language-server-protocol/specifications/specification-3-14/#textDocument_documentSymbol)? And if so, which column has to be ticked of for a given implementation?

Thank you once again!
Nils

**Comment by keegancsmith** [comment]
> Hi @keegancsmith thank you for your swift answer!
> I probably did not phrase that question the right way, sorry for that.
> Let me try again:
> 
> On https://langserver.org/#implementations-server you have a table that lists language server implementations and the functionalities these implementations support.
> Can I infer from that table which servers are likely to support the
> [Document Symbol Request](https://microsoft.github.io/language-server-protocol/specifications/specification-3-14/#textDocument_documentSymbol)? And if so, which column has to be ticked of for a given implementation?
> 
> Thank you once again!
> Nils

we don't currently list it in the table. @chrismwendt @efritz what do you think of adding it in? Important enough of a feature?

**Comment by chrismwendt** [comment]
"Outline view" is currently listed in the "Additional capabilities" column for multiple language servers, so there's a precedent for not making it an explicit column. I haven't seen enough of this kind of feedback to make a decision based on popularity, and the table is already pretty cramped, so I'd lean towards leaving it as-is.

**Comment by nilsmahlstaedt** [comment]
Thank you for your swift answers!

**Comment by goyalyashpal** [comment]
can this issue be closed then?

**Comment by nilsmahlstaedt** [comment]
Hi,
Yeah this was some time ago. I think i found a was around IT somehow.
Thanks for the follow up!



goyalyashpal ***@***.***> schrieb am Mo., 20. März 2023,
18:20:
